app.py

```diff
@@ -39,43 +40,60 @@
 
 # Setup Logging
 logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"),
-                    format='%(levelname)s: %(message)s')
+                    format='[%(asctime)s] %(levelname)s: %(message)s',
+                    datefmt='%m/%d/%y %H:%M:%S')
 
 # Global variables
 static_img_path = "static/img/images/"
 temp_img_prefix = "MAX-"
 image_captions = collections.OrderedDict()
-VALID_EXT = ['png', 'jpg', 'jpeg']
+VALID_EXT = ['.png', '.jpg', '.jpeg']
 error_raised = []
+app_cookie = 'max-image-caption-generator-web-app'
 
 
-class MainHandler(web.RequestHandler):
+class BaseHandler(web.RequestHandler):
+    def prepare(self):
+        # Documentation on cookies:
+        # http://www.tornadoweb.org/en/stable/guide/security.html
+        if not self.get_cookie(app_cookie):
+            user_id = str(uuid.uuid4())
+            self.set_cookie(app_cookie, user_id)
+            logging.info('New user cookie set: ' + user_id)
+        else:
+            logging.info('User cookie found: '
+                         + self.get_cookie(app_cookie))
+
+
+class MainHandler(BaseHandler):
     def get(self):
-        self.render("index.html", image_captions=image_captions)
+        clean_up_old_images()
+        cookie = self.get_cookie(app_cookie)
+        self.render("index.html", image_captions=get_image_captions(cookie))
```

Comment:
You'll want to add a warning about cookies to the bottom of the page for EU users; see https://www.howtogeek.com/327268/why-do-some-websites-have-pop-up-warnings-about-cookies/

You can use the cookie to store whether the user has agreed to your use of cookies.

Reply:
Hmmm, I'll look into this, if I prompt the user and they say no then I'd have to disable upload.

Reply:
I'd say that just disabling the entire app if they say no is fine.

Reply:
In that case that'll be easy to implement, I can include it in my BaseHandler